src/sonic-config-engine/sonic-cfggen

```diff
@@ -221,13 +223,18 @@ def main():
 
     data = {}
     hwsku = args.hwsku
+    asic_name = args.namespace
+    asic_id = None
+    if asic_name is not None:
+        asic_id = asic_name[-1]
+        asic_role = parse_asic_sub_role(args.minigraph if args.minigraph else '/etc/sonic/minigraph.xml', asic_name)
```

Comment:
why do you need specify /etc/sonic/minigraph.xml, it is in line 196.

Reply:
This is needed when the sonic-cfggen command is executed without -m option.

Reply:
Should we show error in this case?

Reply:
There are a few platform related methods which execute 'sonic-cfggen -H' to get the platform data and other information from device_metadata. 
To support these cases for multi-npu platforms, the miigraph from the default location will be used if no minigraph is provided.

Reply:
I think In this case you need to use option --minigraph= and provide the required minigraph

Reply:
Changed to return error when --minigraph is not provided when trying to generate platform info for specific asic.